run/system-package/main_test.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2019 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE:
+# These unit tests mock subprocess.run
+
+import main
+import mock
+import pytest
+
+
+class SubprocessOject:
+    def __init__(self, data=None):
+        self.stdout = data
+
+
+@pytest.fixture
+def client():
+    main.app.testing = True
+    return main.app.test_client()
+
+
+def test_empty_query_string(client):
+    r = client.get('/diagram.png')
+    assert r.status_code == 400
+
+
+def test_empty_dot_parameter(client):
+    r = client.get('/diagram.png?dot=')
+    assert r.status_code == 400
+
+
+@mock.patch('main.subprocess.run',
+            mock.MagicMock(return_value=SubprocessOject()))
+def test_bad_dot_parameter(client):
+    r = client.get('/diagram.png?dot=digraph')
+    assert r.status_code == 400
+
+
+@mock.patch('main.subprocess.run',
+            mock.MagicMock(return_value=SubprocessOject('Image')))
+def test_good_dot_parameter(client):
+    r = client.get(
+        '/diagram.png?dot=digraph G { A -> {B, C, D} -> {F} }')
+    assert r.status_code == 200
```

Comment:
It's great that we have tests, however this is not a strong assertion. Our program could do nothing and still respond with a `200`.

Can we test things like the `Content-Type` and the content of the response is not empty or has some value?

Reply:
This is hard b/c I don't have the package installed on my computer - so I wanted to mock the package.

Reply:
The way this has been handled in Java, Go, and Node is to add the graphviz package to teh containers used for testing.

Reply:
I see - so far all the python tests for the other samples I've worked on could be run locally without any special dependencies.  @grant Do you know how to include the package for testing?

Reply:
>  @grant Do you know how to include the package for testing?

Add `graphviz==0.13.2` to `requirements.txt`?

I don't know how the other languages test, but we should just follow their tests, but for python.

Reply:
The issue there is that would be a python module.  The point is that it's a system package installed in the container - it should run a binary.

Reply:
Sorry, I really don't know what we mean here. Maybe ask @grayside since he seems to know what is being asked and wrote the other samples?

Reply:
Talked to Averi and she said she installed graphviz locally for Java , so I did the same for Python.  It has a different location path when you run brew install vs how it's installed in the container, but I put the path in a global variable to easily be able to switch for local tests.

Reply:
Currently blocked by https://github.com/googleapis/testing-infra-docker/pull/45

Reply:
I believe graphviz is now available in the published image. Can a stronger assertion be made here now? Or is this thread irrelevant now?

Reply:
Updated!